pkg/operator/encryption/controllers/state_controller.go

```diff
@@ -103,6 +105,78 @@ func (c *stateController) sync() error {
 	return configError
 }
 
+type eventWithReason struct {
+	reason  string
+	message string
+}
+
+// eventsFromEncryptionConfigChanges return slice of event reasons with messages corresponding to a difference between current and desired encryption config.
```

Comment:
nit: I think we call it (`map[schema.GroupResource]state.GroupResourceState`) an encryption state.